tracing-appender/src/rolling.rs

```diff
@@ -67,20 +114,99 @@ pub fn hourly(
     RollingFileAppender::new(Rotation::HOURLY, directory, file_name_prefix)
 }
 
+/// Creates an daily, rolling file appender.
+///
+/// The appender returned by `rolling::daily` can be used with `non_blocking` to create
+/// a non-blocking, daily file appender.
+///
+/// The location of the log file will be specified the `directory` passed in.
+/// `file_name_prefix` specifies the prefix of the log file. It is appended by the current date.
+///
+/// # Examples
+///
+/// ``` rust
+/// # #[clippy::allow(needless_doctest_main)]
+/// fn main () {
+///
+/// # fn doc() {
+///     let appender = tracing_appender::rolling::daily("/some/path", "rolling.log");
+///     let (non_blocking_appender, _guard) = tracing_appender::non_blocking(appender);
+///
+///     let subscriber = tracing_subscriber::fmt().with_writer(non_blocking_appender);
+///
+///     tracing::subscriber::with_default(subscriber.finish(), || {
+///         tracing::event!(tracing::Level::INFO, "Hello");
+///     });
+/// # }
+/// }
+/// ```
 pub fn daily(
     directory: impl AsRef<Path>,
     file_name_prefix: impl AsRef<Path>,
 ) -> RollingFileAppender {
     RollingFileAppender::new(Rotation::DAILY, directory, file_name_prefix)
 }
 
+/// Creates a non-rolling, file appender
+///
+/// The appender returned by `rolling::never` can be used with `non_blocking` to create
+/// a non-blocking, appender that never rotates.
+///
+/// The location of the log file will be specified the `directory` passed in.
+/// `file_name_prefix` specifies the prefix of the log file. No date is appended to it.
+///
+/// # Examples
+///
+/// ``` rust
+/// # #[clippy::allow(needless_doctest_main)]
+/// fn main () {
+///
+/// # fn doc() {
+///     let appender = tracing_appender::rolling::never("/some/path", "non-rolling.log");
+///     let (non_blocking_appender, _guard) = tracing_appender::non_blocking(appender);
+///
+///     let subscriber = tracing_subscriber::fmt().with_writer(non_blocking_appender);
+///
+///     tracing::subscriber::with_default(subscriber.finish(), || {
+///         tracing::event!(tracing::Level::INFO, "Hello");
+///     });
+/// # }
+/// }
+/// ```
 pub fn never(
     directory: impl AsRef<Path>,
     file_name_prefix: impl AsRef<Path>,
 ) -> RollingFileAppender {
     RollingFileAppender::new(Rotation::NEVER, directory, file_name_prefix)
 }
 
+/// Defines a fixed period for rolling of a log file.
+///
+/// To use a `Rotation`, you can do the following:
+///
+/// To get an hourly rotation:
```

Comment:
```suggestion
/// ### Hourly Rotation
```
(take it or leave it.)